tests/infrastructure/vm_console_proxy/utils.py

```diff
@@ -66,3 +70,27 @@ def get_vm_console_proxy_resource(resource_kind, namespace=None):
             name=f"{TOKEN_API_VERSION}.{TOKEN_ENDPOINT}" if resource_kind == APIService else VM_CONSOLE_PROXY
         )
     return vm_console_proxy_resource_object
+
+
+def assert_resource_status_and_wait_for_availability(resource: Any, timeout: int) -> None:
+    assert resource.exists, (
+        f"Missing: {resource.kind}/{resource.name} under {getattr(resource, 'namespace', 'cluster')}"
+    )
+
+    valid_kinds = {"APIService", "Service", "Deployment"}  # valid kinds that require an availability check
+    if resource.kind not in valid_kinds:
+        LOGGER.info(f"Resource {resource.kind} is not in the list of valid resources for availability check. Skipping.")
+        return
+
+    if not hasattr(resource, "status"):
```

Comment:
Can it not have status?

Reply:
we expected that it will have but in case there is a bug, we can catch it here

Reply:
why would resource.instance.status not catch the same bug? Can't it take a few seconds to populate status? Are you sure that would be a bug?